apps/api/src/services/programs.ts

```diff
@@ -36,25 +39,43 @@ export class ProgramsService {
       .from(majorSpecialization)
       .innerJoin(major, eq(majorSpecialization.id, major.id))
       .innerJoin(degree, eq(major.degreeId, degree.id));
+
+    if (query.majorId) {
+      majorBuilder.where(eq(major.id, query.majorId));
```

Comment:
Queries are immutable, meaning this line creates a new query with your `.where` and immediately drops it. The solution is to instead do an unconditional `.where` and, inside, spread the result of a ternary which checks the query. In code, something like:
`.where(...query.majorId ? [eq(major.id, query.majorId)] : [])`

Do not use `let` and reassign; it is generally bad practice to introduce `let` unnecessarily.